tls/conn.go

```diff
@@ -19,6 +19,8 @@ import (
 	"time"
 )
 
+var HandshakeFailureErr = &net.OpError{Op: "remote error", Err: alert(alertHandshakeFailure)}
```

Comment:
If we want to export the form of the error (mainly the problem just being the opaque `alertHandshakeFailure`), we can do so in one of the helper files. The actual comparison may or may not need to be done using [reflect.DeepEqual](https://golang.org/pkg/reflect/#DeepEqual).